src-tauri/src/main.rs

```diff
@@ -1,33 +1,47 @@
 // Prevents additional console window on Windows in release, DO NOT REMOVE!!
 #![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]
 
+/**
+ * @TODO
+ * figure out how to only show logs in dev without
+ * commenting out code before push.
+ */
 mod error;
 
+// use log::info;
 use error::Error;
-use fs_extra::dir::{self};
+use fs_extra::dir;
 use nmw::FolderStat;
 use rayon::prelude::*;
 use specta::collect_types;
 use tauri_specta::ts;
 use tokio::sync::oneshot;
 
+// use tauri_plugin_log::LogTarget;
+
 #[tauri::command]
 #[specta::specta]
 async fn get_dir_data(pattern: &str) -> Result<Vec<FolderStat>, Error> {
     let (tx, rx) = oneshot::channel::<Vec<FolderStat>>();
-    let files = nmw::get_file_names(pattern);
+    let dirs = nmw::get_dir_names(pattern);
     let mut result = vec![];
```

Comment:
```suggestion
    let mut result = Vec::with_capacity(dirs.len());
```

By using `with_capacity` here we can instruct `Vec` to allocate space for all `FolderStat` entries upfront (this is possible bc we know the exact number of entries we will have at this point). This reduces overhead during the `collect_into_vec` phase since the `Vec` never needs to copy itself to a new memory location to make space for new elements.